src/asm/output.c

```diff
@@ -287,62 +310,29 @@ static void addexports(void)
 	}
 }
 
-/*
- * Allocate a new patchstructure and link it into the list
- */
-struct Patch *allocpatch(void)
+static void writerpn(uint8_t *rpnexpr, uint32_t *rpnptr_, uint8_t *rpn,
+		     uint32_t rpnlen)
 {
-	struct Patch *pPatch;
-
-	pPatch = malloc(sizeof(struct Patch));
-
-	if (pPatch == NULL)
-		fatalerror("No memory for patch");
-
-	pPatch->pNext = pCurrentSection->pPatches;
-	pPatch->nRPNSize = 0;
-	pPatch->pRPN = NULL;
-	pCurrentSection->pPatches = pPatch;
-
-	return pPatch;
-}
-
-/*
- * Create a new patch (includes the rpn expr)
- */
-void out_CreatePatch(uint32_t type, struct Expression const *expr)
-{
-	struct Patch *pPatch;
-	uint8_t *rpnexpr;
+#define rpnptr (*rpnptr_) /* Oh, how I miss C++ references... */
```

Comment:
This is generally a very bad idea for readability.

Reply:
I went with a different approach. Is it better?

Reply:
Somewhat, but still a bad idea. Same with `popbyte`, really, although I can't blame you for that one. Hidden side effects are extremely unexpected in C, and thus are a terrible idea — C++ coders might find them natural given the overabundance of """smart""" features in the language, but they don't exist in C, and C coders don't expect them.

Reply:
`writebyte` is called as a function, so you should expect side effects...

Reply:
...into local variables that are not passed as arguments?

Reply:
The arguments are implicit because they're part of how this function works, the macro def is simple and right above the code you're reading, and making the arguments explicit would be very redundant and improve nothing.